nexusdialog/src/main/java/com/github/dkharrat/nexusdialog/controllers/LabeledFieldController.java

```diff
@@ -56,7 +73,26 @@ public String getLabel() {
      *                  field can be empty.
      */
     public void setIsRequired(boolean required) {
-        this.required = required;
+        if (! required) {
+            // Multiple instances of the validator may be present.
+            Iterator<InputValidator> iterator = validators.iterator();
+            while (iterator.hasNext()){
+                if(iterator.next() instanceof RequiredFieldValidator) {
+                    iterator.remove();
+                }
+            }
+        } else if (! isRequired()) {
+            validators.add(new RequiredFieldValidator());
```

Comment:
That's a good point. Here's another approach:

Create a single `RequiredFieldValidator` instance in `LabeledFieldController` in a private/protected variable. Use that instance in this method to add or remove it from the validators set. With this approach, you never need to worry about adding multiple instances and you can also remove it easily without having to iterate through the set.